### PR TITLE
Monitorformaladditions

### PR DESCRIPTION
**Pull request by brrmorre**
**Related issue**: 

**Type of change**: feature request

**Impact**: API modification

**Development Phase**: implementation

**Release Notes**
This adds utilities that can be used for formal.

### CODE REVIEW

File: src/main/scala/formal/FormalUtils.scala [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      case _ => ""
+    }
+    val proposed_src = prop_type.toString + "_" + line_info
+
+//    var max_rand = 15
+//    val random_src = scala.util.Random
+//    while (!prop_name_set.add(proposed_src)) {
+//      proposed_src = src + "_" + "%x".format(random_src.nextInt(max_rand))
+//      max_rand = max_rand * 2
+//    }
```
Comment (hcook) [inline src/main/scala/formal/FormalUtils.scala]: Can you delete this commented out code? (FYI using `scala.util.Random` means your tests wont be reproducible from a known seed).

File: src/main/scala/formal/FormalUtils.scala [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            prop_type: PropertyClass)(implicit sourceInfo: SourceInfo): Unit = {
+    Property(dir, cond, crosscond, message, prop_type, "")
+  }
+  def apply(cond: Bool)(implicit sourceInfo: SourceInfo): Unit ={
+    Property(MonitorDirection.Monitor, cond, "Sanity Property", PropertyClass.LocalRTL, "")
+  }
+}
+
+object SourceGet {
+  def get_line_num(implicit sourceInfo: SourceInfo): String = {
```
Comment (hcook) [inline src/main/scala/formal/FormalUtils.scala]: camelCase is the preferred style for scala methods (as opposed to chisel wires).

Comment (brrmorre) [reply src/main/scala/formal/FormalUtils.scala]: I'm moving a file from federation to rocket chip, this cannot be changed without changing the corresponding calls (in federation).

**Comment by brrmorre** [comment]
> Can you please fill out the PR message and give it a nice title. This is a public repository.
> 
> Will all the converted monitor conditions still fire in simulation as they would have when they were assertions?

The default behavior will not change. The assume calls default to normal asserts. They only change if a default parameter is changed.

**Pull request merged**